// @hisptz/ngx-dhis2-menu/esm2015/lib/containers/menu/menu.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { MenuService } from '../../services/menu.service';
import { SystemStateService } from '../../services/system-state.service';
import { MENU_BACKGROUND_COLORS } from '../../constants/menu-background-colors';
export class MenuComponent {
    /**
     * @param {?} menuService
     * @param {?} systemStatusService
     */
    constructor(menuService, systemStatusService) {
        this.menuService = menuService;
        this.systemStatusService = systemStatusService;
        this.rootUrl = '../../../';
        this.menuLoading = true;
        this.menuLoadingFail = false;
        this.loggedIn$ = this.systemStatusService.getLoginStatus();
        this.online = false;
        this.menuNotification = '';
        this.wasOffline = true;
        this.backgroundColor = '#ECECEC';
        this.showSidebar = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.systemStatusService.checkOnlineStatus().subscribe((onlineStatus) => {
            this.online = onlineStatus;
            if (this.online && this.wasOffline) {
                this.menuNotification = 'You are online';
                this.wasOffline = false;
                /**
                         * Hide notification status after sometimes
                         */
                setTimeout(() => {
                    this.menuNotification = '';
                }, 3000);
                /**
                         * Load system settings if failed
                         */
                if (this.menuLoadingFail) {
                    this.menuLoading = true;
                    this.getSystemSettings();
                }
            }
            else if (!this.online) {
                this.menuNotification = 'You are offline';
                this.wasOffline = true;
            }
        });
        this.getSystemSettings();
    }
    /**
     * @param {?} e
     * @return {?}
     */
    toggleSideBar(e) {
        e.stopPropagation();
        this.showSidebar = !this.showSidebar;
    }
    /**
     * @return {?}
     */
    getSystemSettings() {
        this.menuService.getSystemSettings(this.rootUrl).subscribe((settings) => {
            if (settings) {
                this.applicationTitle = settings['applicationTitle'];
                // get system current background style
                const /** @type {?} */ colorName = settings.hasOwnProperty('currentStyle')
                    ? settings['currentStyle'].split('/')[0]
                    : settings.hasOwnProperty('keyStyle')
                        ? settings['keyStyle'].split('/')[0]
                        : 'blue';
                this.backgroundColor =
                    MENU_BACKGROUND_COLORS[colorName];
                this.contextPath = settings.contextPath ? settings.contextPath + '/' : '';
            }
            this.menuLoading = false;
            this.menuLoadingFail = false;
        }, () => {
            this.menuLoading = false;
            this.menuLoadingFail = true;
        });
    }
}
MenuComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-dhis2-menu',
                template: `<div class="menu-container">

  <!-- notification area -->
  <div class="menu-notification-area" *ngIf="menuNotification !== ''" [@open]>
    {{ menuNotification }}
  </div>
  <!-- end of notification area -->
  <!-- login form area -->
  <div class="login-form-container" *ngIf="!(loggedIn$ | async)">
    <app-login-form></app-login-form>
  </div>
  <!-- end of login form area -->

  <!--menu loading area-->
  <div class="menu-loader" *ngIf="menuLoading">
    <app-menu-loader></app-menu-loader>
  </div>
  <!--end of menu loading area-->

  <!--menu loading failed area-->
  <div class="menu-content" *ngIf="!menuLoading && menuLoadingFail" style="background-color: #eeeeee">
    &nbsp;
  </div>
  <!--end of menu loading failed area-->

  <!--menu content area-->
  <div class="menu-content" [style.background-color]="backgroundColor" *ngIf="!menuLoading && !menuLoadingFail">
    <!--menu left block-->
    <div class="menu-left-block">

      <!--menu brand area-->
      <div class="menu-brand">
        <a [href]="rootUrl + 'dhis-web-commons-about/redirect.action'">
          <img [src]="rootUrl + 'api/staticContent/logo_banner'" alt="">
        </a>
      </div>
      <!--end of menu brand area-->

      <!--menu title area-->
      <div class="menu-title-block">
        <a [href]="rootUrl + 'dhis-web-commons-about/redirect.action'" class="menu-title">
          {{applicationTitle}}
        </a>
      </div>
      <!--end of menu title area-->
    </div>
    <!--end of menu left block-->

    <!--menu right block-->
    <div class="menu-right-block">
      <div class="menu-notification-block">
        <app-menu-notification [rootUrl]="rootUrl"></app-menu-notification>
      </div>
      <!--menu search block-->
      <div class="menu-search-block">
        <app-menu-search></app-menu-search>
      </div>
      <!--end of menu search block-->

      <!--menu profile block-->
      <div class="menu-profile-block">
        <app-menu-profile [rootUrl]="rootUrl" [contextPath]="contextPath" [backgroundColor]="backgroundColor"></app-menu-profile>
      </div>
      <!--end of menu profile block-->

      <!--menu sidebar toggle button-->
      <div
        class="menu-toggle-button"
      >
        <a
          (click)="toggleSideBar($event)"
        >
          <svg
            class="menu-toggle-icon"
            viewBox="0 0 24 24"
          >
            <path d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"></path>
          </svg>
        </a>
      </div>
      <!--end of menu sidebar toggle button-->

      <!--sidebar menu block-->
      <div class="sidebar-menu-nav" *ngIf="showSidebar">
        <app-menu-side-bar></app-menu-side-bar>
      </div>
      <!--end of sidebar menu block-->
      <div></div>
    </div>
    <!--end of menu right block-->
  </div>
  <!--end of menu content area-->
</div>
`,
                styles: [`.menu-container{position:fixed;left:0;top:0;right:0;z-index:1400;height:48px}.menu-loader{height:100%}.menu-content{display:flex;width:100%;height:100%;justify-content:space-between}.menu-left-block,.menu-right-block{display:flex}.menu-brand{width:155px;height:100%;display:flex;justify-content:center;align-items:center}.menu-brand>a>img{display:block;margin:auto;max-height:44px;max-width:155px}.menu-title-block{overflow:hidden;white-space:nowrap;color:#fff;align-items:center;justify-items:center;display:flex;padding-right:1rem}.menu-title{font-size:1rem;font-weight:700;color:#fff;text-decoration:none;text-overflow:ellipsis;min-width:1px;overflow:hidden;white-space:nowrap}.menu-search-block{height:100%}.login-form-container{position:fixed;z-index:1500;top:2px;left:20%;min-height:42px;border-color:#f0c36d;background-color:#f9edbe;border-radius:3px}.menu-notification-area{position:fixed;z-index:1500;top:7px;left:20%;height:auto;border-color:#f0c36d;background-color:#f9edbe;border-radius:3px;padding:5px;font-size:15px}.menu-toggle-button{display:none;flex-direction:row;justify-content:flex-end;padding-right:10px;padding-top:6px}.menu-toggle-icon{display:inline-block;height:35px;width:35px;fill:#fff}.sidebar-menu-nav{position:absolute;top:100%;width:400px;height:100vh;right:0;z-index:900;font-size:14px;text-align:left;list-style:none;background-color:#fff;-webkit-background-clip:padding-box;transition:width .5s cubic-bezier(.39,-.04,.2,1);background-clip:padding-box;box-shadow:0 6px 12px rgba(0,0,0,.175)}@media (max-width:400px){.sidebar-menu-nav{width:95%}}@media (max-width:500px){.login-form-container{top:50px;left:2%;width:96%;height:auto}.menu-notification-area{left:2%;right:2%;width:96%;height:auto}}@media (max-width:999px){.menu-profile-block,.menu-search-block,.menu-title-block{display:none}.menu-brand{justify-content:flex-start;margin-left:15px}.menu-toggle-button{display:flex}}.menu-notification-block{display:flex;align-items:center;height:100%;margin-right:30px}@media (min-width:1010px){.sidebar-menu-nav{display:none!important}}`],
                animations: [
                    trigger('open', [
                        state('in', style({
                            opacity: 1
                        })),
                        transition('void => *', [
                            style({
                                opacity: 0
                            }),
                            animate(700)
                        ]),
                        transition('* => void', [
                            animate(400),
                            style({
                                opacity: 0
                            })
                        ])
                    ])
                ]
            },] },
];
/** @nocollapse */
MenuComponent.ctorParameters = () => [
    { type: MenuService },
    { type: SystemStateService }
];
function MenuComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    MenuComponent.prototype.rootUrl;
    /** @type {?} */
    MenuComponent.prototype.applicationTitle;
    /** @type {?} */
    MenuComponent.prototype.backgroundColor;
    /** @type {?} */
    MenuComponent.prototype.menuLoading;
    /** @type {?} */
    MenuComponent.prototype.menuLoadingFail;
    /** @type {?} */
    MenuComponent.prototype.loggedIn$;
    /** @type {?} */
    MenuComponent.prototype.online;
    /** @type {?} */
    MenuComponent.prototype.menuNotification;
    /** @type {?} */
    MenuComponent.prototype.wasOffline;
    /** @type {?} */
    MenuComponent.prototype.showSidebar;
    /** @type {?} */
    MenuComponent.prototype.contextPath;
    /** @type {?} */
    MenuComponent.prototype.menuService;
    /** @type {?} */
    MenuComponent.prototype.systemStatusService;
}
