// @hisptz/ngx-dhis2-menu/esm2015/lib/components/menu-search/menu-search.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { animate, state, style, transition, trigger } from '@angular/animations';
import { MenuService } from '../../services/menu.service';
export class MenuSearchComponent {
    /**
     * @param {?} menuService
     */
    constructor(menuService) {
        this.menuService = menuService;
        this.rootUrl = '../../../';
        this.searchWidth = 47;
        this.showApps = false;
        this.apps = [];
        this.originalApps = [];
        this.loadingModules = true;
        this.filteredApp = '';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.menuService
            .getMenuModules(this.rootUrl)
            .subscribe((menuModules) => {
            if (menuModules !== null) {
                this.loadingModules = false;
                this.originalApps = [...menuModules];
                this.apps = this._prepareMenuModules();
            }
        });
    }
    /**
     * @return {?}
     */
    _prepareMenuModules() {
        return this.filteredApp === ''
            ? this.originalApps.filter((menu) => {
                return !menu.onlyShowOnSearch;
            })
            : this.originalApps;
    }
    /**
     * @return {?}
     */
    updateMenuModules() {
        this.apps = this._prepareMenuModules();
    }
    /**
     * @param {?=} e
     * @return {?}
     */
    widdenSearch(e) {
        if (e) {
            e.stopPropagation();
        }
        document.getElementById('menu_search_input').focus();
        this.showApps = true;
    }
    /**
     * @param {?=} e
     * @return {?}
     */
    reduceSearch(e) {
        if (e) {
            e.stopPropagation();
        }
        document.getElementById('menu_search_input').blur();
        this.showApps = false;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    toggleSearch(e) {
        e.stopPropagation();
        if (this.showApps) {
            this.reduceSearch();
        }
        else {
            this.widdenSearch();
        }
    }
}
MenuSearchComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-menu-search',
                template: `<div class="menu-search-container" appClickOutside (clickOutside)="reduceSearch()">
  <div class="menu-search-icon">
    <svg class="" viewBox="0 0 24 24"
         style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: white; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; padding: 0px;">
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"></path>
    </svg>
  </div>
  <div class="menu-search-input-block">
    <input type="text" class="menu-search-input" id="menu_search_input" placeholder="Search Apps"
           (focus)="widdenSearch($event)" [(ngModel)]="filteredApp" (ngModelChange)="updateMenuModules()"/>
    <span class="underline"></span>
  </div>
  <div class="menu-toggle-link">
    <a (click)="toggleSearch($event)">
      <svg class="menu-toggle-icon" viewBox="0 0 24 24">
        <path
          d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"></path>
      </svg>
    </a>
  </div>

  <div class="menu-list-block" [@slide] *ngIf="showApps">
    <div *ngIf="(apps | filterByName: filteredApp).length !== 0" class="menu-inner-list-block">
      <a *ngFor="let app of  apps | filterByName:filteredApp" [href]="app.defaultAction" class="app-menu-btn"
         title="{{ app.description }}">
        <div class="app-menu-inner-block">
          <img alt="" [src]="app.icon">
          <div class="app-menu-name">{{app.displayName}}</div>
        </div>
      </a>
    </div>
    <div class="menu-inner-list-block-no-match" *ngIf="(apps | filterByName: filteredApp).length == 0">
      <p>No match found!</p>
    </div>
    <div class="menu-app-management-block">
      <a [href]="rootUrl + 'dhis-web-menu-management/'" class="app-management-btn">
        Manage my Apps
      </a>
    </div>
  </div>
</div>
`,
                styles: [`.menu-search-container{display:flex;position:relative;height:100%}.menu-search-input-block{width:390px}#menu_search_input{padding:0!important;margin:0!important;background-color:transparent;color:#fff!important;float:right;border-style:none;border-bottom:1px solid #f1f1f1;transition:all .2s ease-in-out;display:flex;align-items:center;height:37px;font-size:15px;width:100%}#menu_search_input::-webkit-input-placeholder{color:rgba(255,255,255,.7)}#menu_search_input::-moz-placeholder{color:rgba(255,255,255,.7)}#menu_search_input:-ms-input-placeholder{color:rgba(255,255,255,.7)}#menu_search_input:-moz-placeholder{color:rgba(255,255,255,.7)}#menu_search_input:focus{outline:0}#menu_search_input:focus+.underline{-webkit-transform:scale(1);transform:scale(1)}.underline{background-color:#fff;position:absolute;height:2px;width:390px;bottom:11px;-webkit-transform:scale(0,1);transform:scale(0,1);transition:all .3s linear}.menu-toggle-link{margin:10px}.menu-toggle-link a{cursor:pointer}.menu-toggle-icon{display:block;height:25px;width:25px;fill:#fff}.menu-list-block{position:absolute;display:block;top:49px;width:590px;min-height:430px;right:10px;z-index:-1000;font-size:14px;text-align:left;list-style:none;background-color:#fff;box-shadow:0 0 3px rgba(0,0,0,.3)}.menu-inner-list-block{display:flex;overflow:hidden;flex-wrap:wrap;height:380px;padding:10px 5px 10px 10px}.menu-inner-list-block:hover{overflow:auto}.menu-inner-list-block-no-match{display:flex;overflow:hidden;flex-wrap:wrap;height:380px;font-size:26px;color:#ccc}.menu-inner-list-block-no-match p{padding-top:100px;width:100%!important;text-align:center}.apps{padding:15px;overflow:auto;right:35px;width:600px;height:400px}.apps-hidden{display:none}.apps,.profile{position:absolute;background-color:#fff;border:1px solid #ececec;top:50px;z-index:1000;box-shadow:0 5px 10px rgba(0,0,0,.175)}.app-menu-btn{margin:5px;text-decoration:none;height:120px;width:120px;display:flex;justify-content:center;align-items:center;transition:all .2s ease-in-out;padding:5px}.app-menu-btn:hover{background-color:#f1f1f1;-webkit-transform:scale(1.01);transform:scale(1.01)}.app-menu-btn img{display:block;margin:auto;height:53px}.app-menu-name{margin-top:5px;font-size:13px;text-align:center;color:#222}.menu-inner-list-block::-webkit-scrollbar{width:10px}.menu-inner-list-block::-webkit-scrollbar-track{background-color:rgba(0,0,0,.02)}.menu-inner-list-block::-webkit-scrollbar-thumb{background-color:rgba(0,0,0,.1)}.menu-app-management-block{width:100%;height:50px;display:flex;align-items:center;justify-content:center;box-shadow:0 -1px 2px -1px rgba(0,0,0,.16)}.app-management-btn{transition:all .2s ease-in-out;display:flex;align-items:center;justify-content:center;font-size:13px;height:40px;width:97%;color:#222;text-decoration:none}.app-management-btn:hover{background-color:#eee;-webkit-transform:scale(1.01);transform:scale(1.01)}.menu-search-icon{display:flex;align-items:center;margin-right:7px}`],
                animations: [
                    trigger('slide', [
                        state('in', style({
                            opacity: 1
                        })),
                        transition('void => *', [
                            style({
                                transform: 'translateY(-100%)'
                            }),
                            animate(300)
                        ]),
                        transition('* => void', [
                            animate(300),
                            style({
                                transform: 'translateY(-100%)'
                            })
                        ])
                    ])
                ]
            },] },
];
/** @nocollapse */
MenuSearchComponent.ctorParameters = () => [
    { type: MenuService }
];
MenuSearchComponent.propDecorators = {
    rootUrl: [{ type: Input }]
};
function MenuSearchComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    MenuSearchComponent.prototype.rootUrl;
    /** @type {?} */
    MenuSearchComponent.prototype.searchWidth;
    /** @type {?} */
    MenuSearchComponent.prototype.showApps;
    /** @type {?} */
    MenuSearchComponent.prototype.apps;
    /** @type {?} */
    MenuSearchComponent.prototype.originalApps;
    /** @type {?} */
    MenuSearchComponent.prototype.loadingModules;
    /** @type {?} */
    MenuSearchComponent.prototype.filteredApp;
    /** @type {?} */
    MenuSearchComponent.prototype.menuService;
}
