// @hisptz/ngx-dhis2-menu/esm5/lib/components/menu-profile/menu-profile.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { MenuService } from '../../services/menu.service';
import { PROFILE_MENUS } from '../../constants/profile-menus';
var MenuProfileComponent = /** @class */ (function () {
    function MenuProfileComponent(menuService) {
        this.menuService = menuService;
        this.showProfile = false;
        this.rootUrl = this.contextPath = '../../../';
        this.loadingUser = true;
        this.profileMenus = PROFILE_MENUS;
    }
    /**
     * @return {?}
     */
    MenuProfileComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.menuService.getUserInfo(this.rootUrl).subscribe(function (profile) {
            if (profile) {
                _this.currentUser = {
                    name: profile.displayName,
                    email: profile.email,
                };
            }
            _this.loadingUser = false;
        });
    };
    /**
     * @param {?} e
     * @return {?}
     */
    MenuProfileComponent.prototype.showMenuProfile = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        e.stopPropagation();
        this.showProfile = true;
    };
    /**
     * @param {?=} e
     * @return {?}
     */
    MenuProfileComponent.prototype.hideMenuProfile = /**
     * @param {?=} e
     * @return {?}
     */
    function (e) {
        if (e) {
            e.stopPropagation();
        }
        this.showProfile = false;
    };
    MenuProfileComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-menu-profile',
                    template: "<div class=\"menu-profile-container\" appClickOutside (clickOutside)=\"hideMenuProfile()\">\n    <div class=\"menu-profile-btn\">\n        <a class=\"menu-profile-icon\" [style.background-color]=\"backgroundColor | convertToLighterColor\" (mouseenter)=\"showMenuProfile($event)\">\n            {{currentUser?.name | abbreviate}}\n        </a>\n    </div>\n    <div class=\"profile-menu-list-block\" *ngIf=\"showProfile && !loadingUser\" (mouseleave)=\"hideMenuProfile($event)\">\n        <div class=\"profile-menus-list\">\n            <a *ngFor=\"let profileMenu of profileMenus\" [href]=\"rootUrl + profileMenu.defaultAction\" class=\"profile-menu-btn\">\n                <div>\n                    <img alt=\"\" [src]=\"contextPath + profileMenu.icon\">\n                    <div class=\"profile-menu-name\">{{profileMenu?.name}}</div>\n                </div>\n            </a>\n        </div>\n        <div class=\"user-profile-block\">\n            <div [style.background-color]=\"backgroundColor | convertToLighterColor\" class=\"user-profile-logo\" *ngIf=\"!loadingUser\">{{currentUser?.name | abbreviate}}</div>\n            <div class=\"user-profile-name\">\n                <span *ngIf=\"!loadingUser\"><b>{{currentUser?.name}}</b></span>\n            </div>\n            <div><small *ngIf=\"!loadingUser\">{{currentUser?.email}}</small></div>\n            <div class=\"logout-btn-block\">\n                <a class=\"logout-btn\" [href]=\"rootUrl + 'dhis-web-commons-security/logout.action'\" *ngIf=\"!loadingUser\">\n            LOG OUT\n          </a>\n            </div>\n        </div>\n    </div>\n</div>\n",
                    styles: [".menu-profile-icon{height:34px;width:34px;border-radius:17px;display:flex;justify-content:center;align-items:center;font-size:15px;cursor:pointer;text-decoration:none;color:#eee!important}.menu-profile-btn{margin-right:2rem;margin-top:5px}.profile-menu-list-block{position:absolute;top:42px;max-width:650px;width:100%;min-height:320px;right:10px;z-index:-1000;font-size:14px;text-align:left;list-style:none;background-color:#fff;display:flex;box-shadow:0 0 3px rgba(0,0,0,.3)}.profile-menus-list{width:70%;display:flex;flex-wrap:wrap;overflow:auto;padding:15px;height:300px}.user-profile-block{position:relative;text-align:center;width:30%;height:320px;padding-top:20px;background-color:#f5f5f5}.user-profile-logo{margin:auto;height:60px;width:60px;display:flex;justify-content:center;align-items:center;border-radius:30px;font-size:30px;color:#eee;font-weight:500}.user-profile-name{margin-top:10px}.logout-btn-block{position:absolute;bottom:20px;color:#555;padding:5px;width:100%}.logout-btn{display:block;text-decoration:none;text-align:center}.logout-btn:hover{color:#222}.profile-menu-btn{margin:5px;text-decoration:none;height:120px;width:120px;display:flex;justify-content:center;align-items:center;transition:all .2s ease-in-out;padding:5px}.profile-menu-btn:hover{background-color:#f1f1f1;-webkit-transform:scale(1.01);transform:scale(1.01)}.profile-menu-btn img{display:block;margin:auto;height:50px}.profile-menu-name{margin-top:5px;font-size:11px;text-align:center;color:#222}"]
                },] },
    ];
    /** @nocollapse */
    MenuProfileComponent.ctorParameters = function () { return [
        { type: MenuService }
    ]; };
    MenuProfileComponent.propDecorators = {
        rootUrl: [{ type: Input }],
        contextPath: [{ type: Input }],
        backgroundColor: [{ type: Input }]
    };
    return MenuProfileComponent;
}());
export { MenuProfileComponent };
function MenuProfileComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    MenuProfileComponent.prototype.rootUrl;
    /** @type {?} */
    MenuProfileComponent.prototype.contextPath;
    /** @type {?} */
    MenuProfileComponent.prototype.backgroundColor;
    /** @type {?} */
    MenuProfileComponent.prototype.showProfile;
    /** @type {?} */
    MenuProfileComponent.prototype.currentUser;
    /** @type {?} */
    MenuProfileComponent.prototype.loadingUser;
    /** @type {?} */
    MenuProfileComponent.prototype.profileMenus;
    /** @type {?} */
    MenuProfileComponent.prototype.menuService;
}
