// @hisptz/ngx-dhis2-menu/esm2015/lib/components/menu-notification/menu-notification.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { MenuNotificationService } from '../../services/menu-notification.service';
import { MenuService } from '../../services/menu.service';
import { filter } from 'rxjs/operators';
import * as _ from 'lodash';
export class MenuNotificationComponent {
    /**
     * @param {?} menuNotificationService
     * @param {?} menuService
     */
    constructor(menuNotificationService, menuService) {
        this.menuNotificationService = menuNotificationService;
        this.menuService = menuService;
        this.notificationMenuLinks = {};
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.menuNotificationService.loadNotification(this.rootUrl).subscribe((notificationResults) => {
            this.menuNotification = notificationResults;
        });
        this.menuService.getSanitizedMenus().pipe(filter((result) => result.length > 0)).
            subscribe((sanitizedMenus) => {
            _.each(_.filter(sanitizedMenus, menuItem => menuItem.name.indexOf('interpretation') !== -1 || menuItem.name.indexOf('messaging') !== -1), menuItem => {
                if (menuItem.name.indexOf('interpretation') !== -1) {
                    this.notificationMenuLinks['interpretation'] = menuItem.defaultAction;
                }
                else if (menuItem.name.indexOf('messaging') !== -1) {
                    this.notificationMenuLinks['message'] = menuItem.defaultAction;
                }
            });
        });
    }
}
MenuNotificationComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-menu-notification',
                template: `<div class="menu-notification" *ngIf="menuNotification">
  <a class="menu-chat-notification"
     *ngIf="notificationMenuLinks.interpretation"
     [href]="notificationMenuLinks.interpretation"
     title="You have {{ menuNotification.unreadInterpretations }} unread interpretations"
  >
    <div *ngIf="menuNotification.unreadInterpretations > 0" class="menu-notification-badge">
      {{menuNotification.unreadInterpretations}}
    </div>
    <svg class="" viewBox="0 0 24 24"
         style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: white; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;">
      <path
        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"></path>
    </svg>
  </a>
  <a class="menu-message-conversation-notification"
     *ngIf="notificationMenuLinks.message"
     [href]="notificationMenuLinks.message"
     title="You have {{ menuNotification.unreadMessageConversations }} unread message conversations"
  >
    <div *ngIf="menuNotification.unreadMessageConversations > 0" class="menu-notification-badge">
      {{menuNotification.unreadMessageConversations}}
    </div>
    <svg class="" viewBox="0 0 24 24"
         style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: white; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;">
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"></path>
    </svg>
  </a>
</div>
`,
                styles: [`.menu-notification{display:flex;align-items:center;justify-content:space-around}.menu-chat-notification{padding-top:5px;margin-right:10px}.menu-message-conversation-notification{padding-top:4px;margin-left:10px}.menu-chat-notification,.menu-message-conversation-notification{position:relative}.menu-notification-badge{display:flex;flex-flow:row wrap;place-content:center;align-items:center;position:absolute;top:-4px;right:-6px;font-weight:500;font-size:9px;width:20px;height:20px;border-radius:50%;background-color:#4ca899;color:#fff;z-index:10}`]
            },] },
];
/** @nocollapse */
MenuNotificationComponent.ctorParameters = () => [
    { type: MenuNotificationService },
    { type: MenuService }
];
MenuNotificationComponent.propDecorators = {
    rootUrl: [{ type: Input }]
};
function MenuNotificationComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    MenuNotificationComponent.prototype.rootUrl;
    /** @type {?} */
    MenuNotificationComponent.prototype.menuNotification;
    /** @type {?} */
    MenuNotificationComponent.prototype.notificationMenuLinks;
    /** @type {?} */
    MenuNotificationComponent.prototype.menuNotificationService;
    /** @type {?} */
    MenuNotificationComponent.prototype.menuService;
}
