// @hisptz/ngx-dhis2-menu/esm5/lib/services/system-state.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { switchMap, tap } from 'rxjs/operators';
import { BehaviorSubject, of, timer } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
var SystemStateService = /** @class */ (function () {
    function SystemStateService(httpClient) {
        this.httpClient = httpClient;
        this._loggingStatus$ = new BehaviorSubject(true);
    }
    /**
     * @return {?}
     */
    SystemStateService.prototype.checkOnlineStatus = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return timer(1000, 30000).pipe(switchMap(function () { return of(navigator.onLine); }), tap(function (onlineStatus) {
            _this._checkLoginStatus(onlineStatus);
        }));
    };
    /**
     * @param {?} isOnline
     * @return {?}
     */
    SystemStateService.prototype._checkLoginStatus = /**
     * @param {?} isOnline
     * @return {?}
     */
    function (isOnline) {
        var _this = this;
        if (isOnline) {
            this.pingServer().subscribe(function (pingResult) {
                _this._loggingStatus$.next(pingResult.loggedIn);
            }, function (error) {
                if (isOnline) {
                    _this._loggingStatus$.next(false);
                }
            });
        }
        else {
            this._loggingStatus$.next(true);
        }
    };
    /**
     * @return {?}
     */
    SystemStateService.prototype.getLoginStatus = /**
     * @return {?}
     */
    function () {
        return this._loggingStatus$.asObservable();
    };
    /**
     * @return {?}
     */
    SystemStateService.prototype.pingServer = /**
     * @return {?}
     */
    function () {
        return this.httpClient.get('../../../dhis-web-commons-stream/ping.action');
    };
    SystemStateService.decorators = [
        { type: Injectable, args: [{ providedIn: 'root' },] },
    ];
    /** @nocollapse */
    SystemStateService.ctorParameters = function () { return [
        { type: HttpClient }
    ]; };
    /** @nocollapse */ SystemStateService.ngInjectableDef = i0.defineInjectable({ factory: function SystemStateService_Factory() { return new SystemStateService(i0.inject(i1.HttpClient)); }, token: SystemStateService, providedIn: "root" });
    return SystemStateService;
}());
export { SystemStateService };
function SystemStateService_tsickle_Closure_declarations() {
    /** @type {?} */
    SystemStateService.prototype._loggingStatus$;
    /** @type {?} */
    SystemStateService.prototype.httpClient;
}
