// @hisptz/ngx-dhis2-menu/esm5/lib/components/login-form/login-form.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Component } from '@angular/core';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
var LoginFormComponent = /** @class */ (function () {
    function LoginFormComponent(httpClient) {
        this.httpClient = httpClient;
        this.credentials = {
            username: undefined,
            password: undefined
        };
        this.loginError = '';
    }
    /**
     * @return {?}
     */
    LoginFormComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @param {?} e
     * @return {?}
     */
    LoginFormComponent.prototype.onSubmit = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        var _this = this;
        e.stopPropagation();
        this.loggingIn = true;
        this.loginError = '';
        var /** @type {?} */ headers = new HttpHeaders().set('Content-Type', 'application/x-www-form-urlencoded');
        this.httpClient.post('../../../dhis-web-commons-security/login.action', 'j_username=' +
            this.credentials.username +
            '&j_password=' +
            this.credentials.password, {
            headers: headers
        }).pipe(catchError(this._handleError)).subscribe(function () {
            _this.loggingIn = false;
        }, function (error) {
            _this.loggingIn = false;
            if (error.status === 200 && error.url.indexOf('login') !== -1) {
                _this.loginError = 'Incorrect username or password';
            }
        });
    };
    /**
     * @param {?} err
     * @return {?}
     */
    LoginFormComponent.prototype._handleError = /**
     * @param {?} err
     * @return {?}
     */
    function (err) {
        var /** @type {?} */ error = null;
        if (err.error instanceof Error) {
            // A client-side or network error occurred. Handle it accordingly.
            error = {
                message: err.error,
                status: err.status,
                statusText: err.statusText,
                url: err.url
            };
        }
        else {
            // The backend returned an unsuccessful response code.
            // The response body may contain clues as to what went wrong,
            error = {
                message: err.error instanceof Object ? err.error.message : err.error ? err.error : err.message,
                status: err.status,
                statusText: err.statusText,
                url: err.url
            };
        }
        return throwError(error);
    };
    LoginFormComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-login-form',
                    template: "<form action=\"\" class=\"login-form\" #loginForm=\"ngForm\">\n  <input class=\"login-form-input\" type=\"text\" required [(ngModel)]=\"credentials.username\" name=\"username\" #username=\"ngModel\"\n    placeholder=\"Enter username\" autofocus tabindex=\"0\">\n  <input class=\"login-form-input\" type=\"password\" required [(ngModel)]=\"credentials.password\" name=\"password\" #password=\"ngModel\"\n    placeholder=\"Enter password\" autofocus tabindex=\"1\">\n  <button class=\"login-form-button\" type=\"submit\" [disabled]=\"(!password.valid && !username.valid) || loggingIn\" (click)=\"onSubmit($event)\"\n    tabindex=\"2\">\n    {{loggingIn ? 'Logging in ...' : 'Login'}}\n  </button>\n</form>\n<div *ngIf=\"loginError !== ''\" class=\"login-error-block\">{{loginError}}</div>",
                    styles: [".login-form{display:flex;padding:3px!important;width:100%}.login-form-input{padding-left:7px;display:flex;align-items:center;justify-content:flex-start;margin-right:3px;height:37px;outline:0;border-radius:2px;border:thin solid #ececec}.login-form-button{height:37px;background-color:#fff;border:thin solid #ececec;border-radius:2px;cursor:pointer}.login-error-block{font-size:14px;color:red;padding:5px}@media (max-width:500px){.login-form{display:block}.login-form input{width:100%;margin-bottom:3px;margin-right:0}.login-form button{width:100%;text-align:center}}"]
                },] },
    ];
    /** @nocollapse */
    LoginFormComponent.ctorParameters = function () { return [
        { type: HttpClient }
    ]; };
    return LoginFormComponent;
}());
export { LoginFormComponent };
function LoginFormComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    LoginFormComponent.prototype.credentials;
    /** @type {?} */
    LoginFormComponent.prototype.loginError;
    /** @type {?} */
    LoginFormComponent.prototype.loggingIn;
    /** @type {?} */
    LoginFormComponent.prototype.httpClient;
}
